api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -49,9 +50,16 @@ message CacheConfig {
     repeated config.route.v3.QueryParameterMatcher query_parameters_excluded = 4;
   }
 
-  // Config specific to the cache storage implementation.
-  // [#extension-category: envoy.http.cache]
-  google.protobuf.Any typed_config = 1 [(validate.rules).any = {required: true}];
+  oneof cache_implementation {
```

Comment:
Unfortunately this breaks [backwards compatibility](https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md#backwards-compatibility).
[Annotation](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md#field-level) will be needed here.

Reply:
Good catch, I always forget about Golang ruining oneof for everyone else.

Changed to just making it a separate field, which is more forward-looking anyway (in that this way is compatible with merge-inheritable config).